### PR TITLE
[WIP] Add Fast hash

### PR DESCRIPTION
**Pull request by TwsThomas**
Add fast hash and the minmax hashing encoder.

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
It seems that the error on travis is a genuine one

**Comment by GaelVaroquaux** [comment]
Closing is favor of #108

**Comment by GaelVaroquaux** [comment]
Oops, I closed the wrong one. Reopening.

File: dirty_cat/test/test_minhash_encoder.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    ngram_range = (2,4)
+    enc = MinHashEncoder(n_components=2)
+    ngrams = enc.get_unique_ngrams(string, ngram_range)
+    assert ngrams == true_ngrams
+        
+def profile_encoder(Encoder, hashing='fast', minmax_hash=False):
+
+    from dirty_cat import datasets
+    employee_salaries = datasets.fetch_employee_salaries()
+    data = pd.read_csv(employee_salaries['path'])
```
Comment (alexis-cvetkov) [inline dirty_cat/test/test_minhash_encoder.py:64]: This function computes the time needed to encode a dataset, but rely on pandas to load it.

**Comment by GaelVaroquaux** [comment]
Good job resuscitating travis!

It seems that codecov is failing in the travis run because it is getting unknown arguments.

**Comment by alexis-cvetkov** [comment]
This PR can be merged I think! Can you review it @GaelVaroquaux ?

File: dirty_cat/minhash_encoder.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
+
```
Comment (GaelVaroquaux) [inline dirty_cat/minhash_encoder.py:1]: Add a small docstring to describe what is in this file (to help other future developers)

**Comment by GaelVaroquaux** [comment]
I had a very quick (and incomplete) look. The new class (the MinHashEncoder) needs to be added to the doc, at the bottom of:
https://github.com/dirty-cat/dirty_cat/blob/master/doc/index.rst

File: build_tools/circle/build_doc.sh [hunk 3]
```diff
@@ -116,7 +116,7 @@ conda update --yes --quiet conda
 # provided versions
 conda create -n $CONDA_ENV_NAME --yes --quiet python="${PYTHON_VERSION:-*}" \
   numpy="${NUMPY_VERSION:-*}" scipy="${SCIPY_VERSION:-*}" \
-  pytest coverage matplotlib="${MATPLOTLIB_VERSION:-*}" sphinx \
+  pytest coverage matplotlib="${MATPLOTLIB_VERSION:-*}" sphinx=* \
```
Comment (TwsThomas) [inline build_tools/circle/build_doc.sh:119]:
```suggestion
  pytest coverage matplotlib="${MATPLOTLIB_VERSION:-*}" sphinx \
```
We do not need this here. I guess it is already fixed.

File: dirty_cat/test/test_minhash_encoder.py (hunk 1 above)
Comment (TwsThomas) [reply dirty_cat/test/test_minhash_encoder.py:66]: Yes indeed. 
It was only to assert performances. 
It do not run with unit test, so we could keep it (or remove it).

**Comment by GaelVaroquaux** [comment]
Hurray!!

G

**Pull request merged**